File: C++/summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ans;
        int sz = nums.size();
        if (sz == 1) {
            ans.push_back(to_string(nums[0]));
        } else if (sz > 1) {
            int start = nums[0], end = nums[0];
            for (int i = 1; i <= sz; i++) {
                if (i < sz && nums[i-1] + 1 == nums[i]) {
                    end = nums[i];
                } else {
                    if (start == end) {
                        ans.push_back(to_string(start));
                    } else {
                        ans.push_back(to_string(start) + "->" + to_string(end));
                    }
                    if (i < sz) {
                        start = end = nums[i];
                    }
                }
            }
        }
        return ans;
    }
};

File: TypeScript/summary-ranges.ts
function summaryRanges(nums: number[]): string[] {
    const ans = []
    let start = nums[0]
    for (let i = 1; i <= nums.length; i++) {
        if (i == nums.length || nums[i] - nums[i-1] > 1) {
            const end = nums[i-1]
            if (start == end) {
                ans.push(`${start}`)
            } else {
                ans.push(`${start}->${end}`)
            }
            start = nums[i]
        }
    }
    return ans
};